core/src/banking_stage/vote_storage.rs

```diff
@@ -71,7 +67,7 @@ impl VoteStorage {
             packets.filter_map(|packet| {
                 LatestValidatorVotePacket::new_from_immutable(
                     packet,
-                    self.vote_source,
+                    VoteSource::Tpu, // incorrect, but this bug has been here w/o issue for a long time.
```

Comment:
to clarify: before gossip votes were never re-inserted since gossip vote thread never process received txs. Now, gossip votes will be processed and if reinserted, they will be reinserted as TPU votes?

Reply:
Correct - gossip thread never re-inserted on retry because only TPU thread processed packets. It would re-insert with TPU as the label always.

Now we just always re-insert with Tpu source. It's not correct, but consistent with previous behavior. Want to fix this separately.